torchao/prototype/galore/kernels/quant.py

```diff
@@ -22,17 +24,18 @@ def _dequant_kernel(
     # rm = tl.max_contiguous(tl.multiple_of(rm % M, BLOCK_M), BLOCK_M)
     # rn = tl.max_contiguous(tl.multiple_of(rn % N, BLOCK_N), BLOCK_N)
     offsets = rm[:, None] * stride_qm + rn[None, :] * stride_qn
+    mask = (rm[:, None] < M) & (rn[None, :] < N)
     tl.static_print(offsets)
     group_offsets = offsets // GROUP_SIZE
     tl.static_print("group_offsets", group_offsets)
-    q_idx = tl.load(q_idx_ptr + offsets)
+    q_idx = tl.load(q_idx_ptr + offsets, mask=mask)
     tl.static_print(q_idx)
     # NOTE: Must upcast q_idx to int32 (q_idx is tl.uint8, which does not work for pointer indexing)
     q_vals = tl.load(qmap_ptr + q_idx.to(tl.int32))
-    absmax = tl.load(absmax_ptr + group_offsets)
+    absmax = tl.load(absmax_ptr + group_offsets)  # TODO: might need to add mask for this load also
 
     dq = q_vals * absmax
-    tl.store(dq_ptr + offsets, dq)
+    tl.store(dq_ptr + offsets, dq, mask=mask)
```

Comment:
@msaroufim This should fix the occasional illegal memory address (IMA) errors we observe.
- For some reasons torch2.5 in CI consistently produce this error (instead of randomly like other CI jobs)
- Running the galore tests alone will not reproduce this error due to CUDA caching allocator - the out-of-bounds addresses might still be valid address
  - To reproduce IMA error locally (without running the whole test suite), I allocate random tensors at the start of the test (e.g. `tensor = [torch.randn(torch.randint(100, 2000, ()).item(), torch.randint(100, 2000, ()).item(), device="cuda") for _ in range(100)]`)
- `tl.load(qmap_ptr + q_idx.to(tl.int32))` will never encounter out-of-bounds since `0 <= q_idx < 256` (due to uint8) and `qmap` always has size 256 (if it is used correctly)
- `tl.load(absmax_ptr + group_offsets)` MAY still face out-of-bounds error I think, but since `absmax` tensor is small, it's very likely that out-of-bound addresses are still valid. To be super safe, we would need to add mask for this one too. (sth like `mask=group_offsets < (M * N // GROUP_SIZE)` should be correct I think)